### ISSUE TITLE
GitHub Releases are broken

### ISSUE THREAD

**Issue by DragonEggBedrockBreaking** [Prospector/ModMenu#413]
For some reason the file size is 261 bytes, and just doesn't load up as a mod...

**Comment by Prospector** [comment]
Where did you download it from? It sounds like the file did not download properly for some reason. Try re-downloading the file.

**Comment by DragonEggBedrockBreaking** [comment]
On GitHub it literally shows as 261 bytes… At least check next time.

![C23C62A9-AD2C-49F3-ACE7-78814F657D9C](https://user-images.githubusercontent.com/68545280/163717814-8e4b51b6-f0de-4eda-bac7-9e521a8acb38.jpeg)

**Comment by haykam821** [comment]
This issue also seems to affect [version 4.0.0 beta 2](https://github.com/TerraformersMC/ModMenu/releases/tag/v4.0.0-beta.2).

**Comment by mckeeverwatts** [comment]
Hmm...weird. The files on both CurseForge and Modrinth are fine, download from there for now. I will have to look into why the GitHub one is broken.

**Comment by Prospector** [comment]
Hmm...weird. The files on both CurseForge and Modrinth are fine, download from there for now. I will have to look into why the GitHub one is broken.

**Comment by jhbuchanan45** [comment]
Could be worth trying actions/upload-artifact@v3 instead of the current v2 to see if it makes a difference. It seems to have randomly broken going from v3.1.0 to v3.1.1 with no changes to our github action script.

**Comment by jhbuchanan45** [comment]
Nevermind, it has nothing to do with that action, since the uploaded artifact in the Action run is correct, it must be an issue with the Gradle github task

**Comment by Jab125** [comment]
its literally just the MANIFEST file

**Comment by DragonEggBedrockBreaking** [comment]
It seems to be fixed in 3.2.1

**Comment by Prospector** [comment]
I still don't know why it happened. Nothing ever changed. Going to reopen until we have more evidence to show it won't happen again.

**Comment by jackassmc** [comment]
This appears to have been a bug on GitHub's end. The jar was built correctly - it was uploaded to Modrinth and Curseforge and remained intact. The GitHub Releases upload in ferry uses the same jar that was successfully uploaded to Modrinth and Curseforge. Maybe GitHub tries to improve the compression and something went wrong?

Since the upload broke by itself and then fixed itself I think it's safe to re-enable the GitHub Releases upload in ferry and hope for the best.

**Comment by jackassmc** [comment]
It's broken again. The Modrinth upload is also broken, hopefully Ferry 2.5 fixes it.

**Comment by jackassmc** [comment]
Modrinth upload is still broken in Ferry 2.5.

**Comment by jackassmc** [comment]
The `github` task is broken because it doesn't depend on the `build` task:

<details>
<summary>https://github.com/dorongold/gradle-task-tree output</summary>

```
./gradlew generateChangelog build publish curseforge github modrinth taskTree --repeat

> Configure project :
Fabric Loom: 0.12.54
Missing lastTag or currentTag parameters, aborting changelog creation
No CURSEFORGE_TOKEN specified
Loom is applying dependency data manually to publications instead of using a software component (from(components["java"])). This is deprecated and will be removed in Loom 0.13.
Not publishing sources jar as it was not found. Use java.withSourcesJar() to fix.

> Task :taskTree

------------------------------------------------------------
Root project 'modmenu'
------------------------------------------------------------

:generateChangelog
No task dependencies

:build
+--- :assemble
|    +--- :jar
|    |    \--- :classes
|    |         +--- :compileJava
|    |         \--- :processResources
|    +--- :remapJar
|    |    +--- :jar
|    |    |    \--- :classes
|    |    |         +--- :compileJava
|    |    |         \--- :processResources
|    |    \--- :prepareRemapJar
|    |         \--- :jar
|    |              \--- :classes
|    |                   +--- :compileJava
|    |                   \--- :processResources
|    \--- :remapSourcesJar
|         \--- :sourcesJar
|              \--- :classes
|                   +--- :compileJava
|                   \--- :processResources
\--- :check
     +--- :test
     |    +--- :classes
     |    |    +--- :compileJava
     |    |    \--- :processResources
     |    \--- :testClasses
     |         +--- :compileTestJava
     |         |    \--- :classes
     |         |         +--- :compileJava
     |         |         \--- :processResources
     |         \--- :processTestResources
     \--- :validateAccessWidener

:publish
No task dependencies

:curseforge
\--- :curseforge308702
     +--- :assemble
     |    +--- :jar
     |    |    \--- :classes
     |    |         +--- :compileJava
     |    |         \--- :processResources
     |    +--- :remapJar
     |    |    +--- :jar
     |    |    |    \--- :classes
     |    |    |         +--- :compileJava
     |    |    |         \--- :processResources
     |    |    \--- :prepareRemapJar
     |    |         \--- :jar
     |    |              \--- :classes
     |    |                   +--- :compileJava
     |    |                   \--- :processResources
     |    \--- :remapSourcesJar
     |         \--- :sourcesJar
     |              \--- :classes
     |                   +--- :compileJava
     |                   \--- :processResources
     \--- :remapJar
          +--- :jar
          |    \--- :classes
          |         +--- :compileJava
          |         \--- :processResources
          \--- :prepareRemapJar
               \--- :jar
                    \--- :classes
                         +--- :compileJava
                         \--- :processResources

:github
No task dependencies

:modrinth
\--- :build
     +--- :assemble
     |    +--- :jar
     |    |    \--- :classes
     |    |         +--- :compileJava
     |    |         \--- :processResources
     |    +--- :remapJar
     |    |    +--- :jar
     |    |    |    \--- :classes
     |    |    |         +--- :compileJava
     |    |    |         \--- :processResources
     |    |    \--- :prepareRemapJar
     |    |         \--- :jar
     |    |              \--- :classes
     |    |                   +--- :compileJava
     |    |                   \--- :processResources
     |    \--- :remapSourcesJar
     |         \--- :sourcesJar
     |              \--- :classes
     |                   +--- :compileJava
     |                   \--- :processResources
     \--- :check
          +--- :test
          |    +--- :classes
          |    |    +--- :compileJava
          |    |    \--- :processResources
          |    \--- :testClasses
          |         +--- :compileTestJava
          |         |    \--- :classes
          |         |         +--- :compileJava
          |         |         \--- :processResources
          |         \--- :processTestResources
          \--- :validateAccessWidener


To see task dependency tree for a specific task, run gradlew <project-path>:<task> <project-path>:taskTree [--depth <depth>] [--with-inputs] [--with-outputs] [--repeat]
Executions of all tasks except for taskTree are skipped. They are used for building the task graph only.
For example, try running gradlew :build :taskTree
Deprecated Loom APIs were used in this build, making it incompatible with future versions of Loom. Use Gradle warning modes to control the verbosity of the warnings.

BUILD SUCCESSFUL in 1s
1 actionable task: 1 executed
```
</details>

Thanks to Arthur from GitHub support for figuring it out.

**Comment by jackassmc** [comment]
This issue is resolved in the latest Ferry 2.5.
